### PR TITLE
Add integration of lessons

### PR DESCRIPTION
**Pull request by DoumanAsh**
Introduces lesson integration for webpack.

Currently it loads lessons from folder `lessons/` and expects format of folder inside as `<numeric order>-<name>`
Inside lesson folder there are expected two files:
* `code.cpp` containing C++ code to extract as raw string.
* `text.adoc` lesson content that will be parsed to HTML.

Lessons can be updated during dev server too (not sure about adding new ones)

There is though existing problem, at least for me, that with lessons integrated any change of JS file triggers error.
Not sure if it is webpack fault or something else.

### CODE REVIEW

**Comment by Leandros** [comment]
Can we define a default page via this? So we write the start page also in asciidoc and it gets included this way?

**Comment by DoumanAsh** [comment]
Not at the moment, but it can be done I guess.
I suppose we can use some special format for it in order to distinguish.

**Comment by DoumanAsh** [comment]
@Leandros I found problem with one of the webpack's loaders `eslint-loader` which runs linter in dev server.
For now I disabled it so that we could continue to work with integrated lessons.

I created issue in `eslint-loader`'s dependency that causes error as it seems to be something internal there.

Meanwhile you can take a look as it is now ready more or less.

File: src/templates/index.pug [hunk 1]
```diff
@@ -50,7 +67,7 @@ body.layout
 
     // Sidebar
     div.sidebar#sidebar.hidden
-        +sidebar_link("Introduction", "Introduction")
-        +sidebar_link("0", "Lesson 0: Getting Started")
+        each lesson in lessons
+            +sidebar_link(lesson.id, `${lesson.num}: ${lesson.name}`)
```
Comment (Leandros) [inline src/templates/index.pug]: We should make sure to properly format `${lesson.num}`, especially given that it's handled as a float. I would like to not have `1.299999` as a lesson by accident, due to some rounding errors (1.3 can not be represented as a float, the closest are `1.2999999523162841796875`  or `1.30000007152557373046875`).

File: webpack/get_lessons.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+            const name = path.basename(dir)
+
+            if (name === 'default') {
+                continue
+            }
+
+            //format: num-chapter_part1-chapter_part2
+            let [num, ...chapter] = name.split('-')
+            num = parseFloat(num)
```
Comment (Leandros) [inline webpack/get_lessons.js]: Can we be sure `parseFloat` of `1.3` returns something which will be over 1.3, for formatting purposes (see also my first comment)?

File: src/templates/index.pug (hunk 1 above)
Comment (Leandros) [inline src/templates/index.pug:71]: We should make sure to properly format `${lesson.num}`, especially given that it's handled as a float. I would like to not have `1.299999` as a lesson by accident, due to some rounding errors (1.3 can not be represented as a float, the closest are `1.2999999523162841796875`  or `1.30000007152557373046875`).

File: webpack.config.js [hunk 3]
```diff
@@ -99,7 +118,11 @@ function html_pug_plug(title, template, opts) {
 }
 
 module.exports.plugins = [
-    html_pug_plug("A Tour of C++", "templates/index.pug"),
+    //Inject lessons index through options.
```
Comment (Leandros) [inline webpack.config.js]: Could we agree on having a space between `//` and the first letter, to stay consistent in our comment style?

File: webpack.config.js [hunk 4]
```diff
@@ -52,11 +53,27 @@ const font_loader = {
...
+    ]
+}
+
 const js_loader = {
     test: /\.jsx?$/,
+    exclude: /(node_modules|bower_components)/,
     use: [
         "babel-loader",
-        "eslint-loader"
+        //Disabled temporarely due to erros from its dependency objest-hash
```
Comment (Leandros) [inline webpack.config.js]: Can we still run eslint as part of `npm start`, but not as part of refreshing inside webpack?

File: src/templates/index.pug (hunk 1 above)
Comment (DoumanAsh) [reply src/templates/index.pug:71]: Um... but `parseFloat('1.3')` produces float `1.3` in node.js
I can continue to store that as strings though if you want to be sure.

File: webpack.config.js (hunk 3 above)
Comment (DoumanAsh) [reply webpack.config.js]: Ok

File: webpack.config.js (hunk 4 above)
Comment (DoumanAsh) [reply webpack.config.js]: I'll check out

File: webpack/get_lessons.js (hunk 2 above)
Comment (DoumanAsh) [reply webpack/get_lessons.js]: My node.js and browser parses it as 1.3, but I can also save string representation if you want to be sure.

**Review by Leandros — APPROVED** [review]
LGTM.

**Pull request merged**